src/components/types/types.ts

```diff
@@ -1,69 +1,55 @@
+import { MutableRefObject } from 'react';
+
 import { AppData, AppSetting } from '@graasp/sdk';
-import { ImmutableCast } from '@graasp/sdk/frontend';
 
 import { QuestionType } from '../../config/constants';
+import { PlotDatum } from 'plotly.js-basic-dist-min';
 
 export type AppDataData = {
   questionId: string;
 };
-export type AppDataQuestionRecord = ImmutableCast<
-  AppData & { data: AppDataData }
->;
+export type AppDataQuestion = AppData & { data: AppDataData };
 
 export type MultipleChoiceAppDataData = AppDataData & {
   choices: string[];
 };
-export type MultipleChoiceAppDataDataRecord =
-  ImmutableCast<MultipleChoiceAppDataData>;
 
 export type TextAppDataData = AppDataData & {
   text: string;
 };
 
-export type TextAppDataDataRecord = ImmutableCast<TextAppDataData>;
-
 export type FillTheBlanksAppDataData = AppDataData & {
   text: string;
 };
 
-export type FillTheBlanksAppDataDataRecord =
-  ImmutableCast<FillTheBlanksAppDataData>;
-
 export type SliderAppDataData = AppDataData & {
   value: number;
 };
 
-export type SliderAppDataDataRecord = ImmutableCast<SliderAppDataData>;
-
-export type AppDataDataRecord = ImmutableCast<
-  | MultipleChoiceAppDataData
-  | TextAppDataData
-  | FillTheBlanksAppDataData
-  | SliderAppDataData
->;
-
 export type AppSettingData = {
   type: string;
   question: string;
   questionId: string;
   explanation?: string;
+  numberOfAttempts?: number;
```

Comment:
I don't know how much you did for this feature, is it okay to remove it for now?

Reply:
Yes I will remove it. It has nothing to do in the PR...